### PR TITLE
[MIG][15.0] hr_contract: migration script

### PR DESCRIPTION
**Pull request by sang250399**
This PR
======
Migration done

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
LGTM. Code review / no test.

**Review by gdgellatly — APPROVED** [review]

**Review by leotranviindoo — APPROVED** [review]

**Review by remi-filament — APPROVED** [review]
Tested OK on migration of our database

File: openupgrade_scripts/scripts/hr_contract/15.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.convert_field_to_html(env.cr, "hr_contract", "notes", "notes")
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/hr_contract/15.0.1.0/pre-migration.py:6]: Please copy the column on pre to preserve the original content, and do this on post.

File: openupgrade_scripts/scripts/hr_contract/15.0.1.0/upgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# NOTHING TO DO: new feature on 15.0
+
+hr_contract  / hr.contract              / notes (text)                  : type is now 'html' ('text')
+# DONE: pre-migration: convert text to html
+
+hr_contract  / hr.contract.type         / name (char)                   : NEW required
+# NOTHING TO DO: new model for new feature on 15.0
+
+hr_contract  / hr.employee              / first_contract_date (date)    : is now stored
+# NOTHING TO DO: computed in load
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/hr_contract/15.0.1.0/upgrade_analysis_work.txt:15]: Please add a TODO saying that if very heavy, compute it by SQL.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Superseded by #3688 as no answer

**Pull request closed without merging**